tests/find_and_replace/find_and_replace_plugin.test.ts

```diff
@@ -697,3 +718,25 @@ describe("replace", () => {
     expect(getActivePosition(model)).toBe("A1");
   });
 });
+
+test("replace don't replace value resulting from array formula", () => {
+  model = new Model();
+  setCellContent(model, "A1", "hell0");
+  setCellContent(model, "A2", "hello");
+  setCellContent(model, "A3", "=1");
+  setCellContent(model, "B1", "=TRANSPOSE(A1:A3)");
+  searchOptions = {
+    matchCase: false,
+    exactMatch: false,
+    searchFormulas: false,
+  };
+  model.dispatch("UPDATE_SEARCH", { toSearch: "hello", searchOptions });
+  model.dispatch("REPLACE_ALL_SEARCH", { replaceWith: "kikou" });
+  const matches = model.getters.getSearchMatches();
+  const matchIndex = model.getters.getCurrentSelectedMatchIndex();
+  expect(matches).toHaveLength(0);
+  expect(matchIndex).toStrictEqual(null);
+  expect(getCellContent(model, "A2")).toBe("kikou");
+  expect(getCellContent(model, "C1")).toBe("kikou");
```

Comment:
this tests passes in 17.0 (I think we should fix it in 17.0 as well). this helpers checks the content of the evaluated cell, but not the content of the core cell.

Reply:
It's totally normal that this test passes in 17.0 as is now, as we don't take array formula spreaded values into account when doing a search/replace, so it won't be replaced. This test doesn't have any sense as a standalone (without the behaviour modification I introduce in this fix) :)